crates/swc_ecma_minifier/src/compress/optimize/inline.rs

```diff
@@ -89,8 +89,42 @@ where
                 return;
             }
 
+            let is_inline_enabled =
+                self.options.reduce_vars || self.options.collapse_vars || self.options.inline != 0;
+
             self.vars.inline_with_multi_replacer(init);
 
+            // We inline arrays partially if it's pure (all elements are literal), and not
+            // modified.
+            // We don't drop definition, but we just inline array accesses with numeric
+            // literal key.
+            //
+            // TODO: Allow `length` in usage.accessed_props
+            if usage.declared
+                && !usage.reassigned()
+                && !usage.mutated
+                && !usage.has_property_mutation
+                && usage.accessed_props.is_empty()
+                && !usage.is_infected()
+                && is_inline_enabled
+            {
+                if let Expr::Array(arr) = init {
+                    if arr.elems.len() < 32
+                        && arr.elems.iter().all(|e| match e {
+                            Some(ExprOrSpread { spread: None, expr }) => match &**expr {
+                                Expr::Lit(..) => true,
+                                _ => false,
+                            },
+                            _ => false,
```

Comment:
Couldn't we still optimize `array[0]` even if `array[1]` is an expression?

Reply:
We currently don't analyze accessed indexes.
And I doubt if such analysis will be useful for real-world codes.